Newsletters: add 110 (2020-08-12)

harding opened the pull request:
- [x] Update releases/RCs on Tuesday @harding 
- [x] Review club @jnewbery @jonatack 
- [x] `Bitcoin Core #18991` @jnewbery 
- [x] `Eclair #1499` @dongcarl 
- [x] Veriphi field report

jonatack commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,129 @@
+release candidates.*
+
+- [LND 0.11.0-beta.rc2][lnd 0.11.0-beta] is a release candidate for the
+  next major version of this project.  It will allows accepting [large
```
will allows -> allows or will allow

jonatack commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,129 @@
+
+- [Bitcoin Core #19620][] prevents Bitcoin Core from re-downloading
+  unconfirmed segwit transactions that attempt to spend non-standard
+  UTXOs.  An non-standard UTXO is one that uses a currently discouraged
```
s/An/A/

jonatack commented:
News110 review club (bip155): e64672470c0148a9ab3e42b6e57da436d5b7504d

jnewbery commented:
I've pushed Jon's review club summary to this branch.

jnewbery commented:
I've pushed my summary of bitcoin core 18991

vasild commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,206 @@
+question below to see a summary of the answer from the meeting.*
+
+[Implement ADDRv2 support (part of BIP155)][review club #19031] is a PR
+([#19031][Bitcoin Core #19031]) by Vasil Dimov that proposes a partial
+implementation of the [BIP155][] `addrv2` P2P message.
```
This gives the impression that the PR partially implements `addrv2` whereas it fully implements `addrv2` (and partially BIP155, for full BIP155 we need TORv3, I2P, CJDNS implemented too, using the `addrv2` messages).

So, maybe reword to: ... proposes a partial implementation of BIP155 - the `addrv2` P2P message which would enable nodes to send and receive TORv3, IP2 and other larger addresses.

jonatack replied:
Maybe the shorter "partial implementation of BIP155." We go into the details further down.

vasild commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,206 @@
+{% include functions/details-list.md
+  q0="How does Bitcoin Core handle network addresses?"
+  a0="Bitcoin Core currently sees all peer addresses as being more or less in a
+      [single][19031 single] IPv6-like space. IPv4 and Tor v2 addresses are
```
What is this syntax `[single][19031 single]`? Where can I see how it is rendered? https://github.com/bitcoinops/bitcoinops.github.io/blob/8c021652cd78cdf190a566cc02b41b107a8ec2d0/_posts/en/newsletters/2020-08-12-newsletter.md looks bad.

jonatack replied:
(explained the link format via irc)

(you can render the document by building and running make preview per the readme and opening localhost:4000)

vasild commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,206 @@
+
+  q1="How are peer addresses stored?"
+  a1="Addresses are saved in the `peers.dat` file in a 16-byte format, which
+      corresponds to the length of IPv6 and Tor v2 addresses; the smaller-sized
```
`s/ and Tor v2//` (TORv2 addresses are 10 bytes)

jonatack replied:
The review club study notes might need to be corrected: "Relevant to PR 19031 is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address."

jonatack replied:
(Yes, there is the 6-byte onion prefix of `0xFD,0x87,0xD8,0x7E,0xEB,0x43`).

harding replied:
Should this instead be `s/and Tor v2/and onioncat-encoded Tor v2/` ?

vasild replied:
Yes, "onioncat-encoded Tor v2" would be correct. Since we also store IPv4 in a similar way, maybe reword to: "which corresponds to the length of IPv6 and other addresses encoded as IPv6".

jonatack replied:
Within the context of this paragraph and the preceding ones, maybe:

"which corresponds to the length of IPv6 addresses; the smaller-sized IPv4 and Tor v2 addresses are therefore padded to fit."

jonatack replied:
(was tempted to link to `CNetAddr::SetLegacyIPv6`, but this code has changed considerably from 0.20 to master and the stable release code isn't as clear about it as it is now).

jonatack replied:
> The review club study notes might need to be corrected

done (for info)

harding commented on _includes/articles/veriphi-segwit-batching.md:
```diff
@@ -0,0 +1,74 @@
+
+__Major takeaways__
+
+Assuming a bitcoin price of $9250.00 (correct at the time of publication):
```
Yeah, but where will it be Wednesday morning?  :-D

harding commented:
Pushed edits for feedback.  Some notable unaddressed feedback:

- Update BTC→USD values to current values (more work than it's worth IMO) https://github.com/bitcoinops/bitcoinops.github.io/pull/449#discussion_r467691581
- Decrease precision of financial numbers (style decision IMO, up to report authors) https://github.com/bitcoinops/bitcoinops.github.io/pull/449#discussion_r467692759
- TorV2 storage in addrman (open question on what edit to make) https://github.com/bitcoinops/bitcoinops.github.io/pull/449#discussion_r468052888

If there are any other significant feedback that I've failed to address, please give me a kick.  Thanks everyone!

jonatack commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+  fingerprint the peer or infer network topology. To prevent that, this PR
+  changes Bitcoin Core's behaviour to cache responses to `getaddr` requests and
+  provide the same response to all peers who request addresses in a 24 hour
+  window.
```
Great description @jnewbery! I wonder if it would be good to mention that it's a rolling window (randomly varying between 21 and 27 hours, if memory serves).

jnewbery replied:
good idea. Added the word "rolling"

jnewbery commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+  an alternative to [backporting the wtxid relay feature][Bitcoin Core
+  #19606], which would also prevent that wasted bandwidth.
+
+- [C-Lightning #3909][] the `listpays` RPC now returns a new
```
This isn't a complete sentence. Consider changing to "C-Lightning 3909 changes the `listpays` RPC to return..."

jonatack commented on _posts/en/newsletters/2020-08-12-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+  a4="Tor v2 is planned to be [deprecated on Sept 15, 2020][19031 v2 deprecate]
+      and obsolete on July 15, 2021 ([schedule][19031 v2 schedule]). Tor v3
+      provides much better security and privacy. It uses longer, 56-character
+      addresses and therefore requires implementation of BIP155 `addrv2`."
```
> "Tor v3 provides much better security and privacy, since it uses longer, 56-character addresses. Using Tor v3 on the Bitcoin network therefore requires implementation of BIP155 `addrv2`."

Much better :+1:

harding commented:
I pushed an edit to the Veriphi report reducing the precision of the the numbers as requested by @jonatack and @jnewbery (thanks!).  Although I do think that it's nicer that way, I don't have a strong opinion about it, so please feel free to revert.

bitschmidty commented:
I conferred with Veriphi folks and they are ok with our judgment on these changes.

jonatack commented on _includes/articles/veriphi-segwit-batching.md:
```diff
@@ -38,24 +38,21 @@ code used [here][veriphi segwit repo].
+- From 24 August 2017 to 30 June 2020, about 37 thousand BTC could have
+  been saved by Bitcoin users if they would have all been using segwit native
+  (bech32), down from over 97 thousand BTC
+  actually paid in fees, which is 38.00% in savings or about $340 million USD.
```
ACK on this commit 6067ddfac22b7dac. If you retouch, consider maybe changing 9.97% to 10% and 38.00% to 38%.

jonatack commented:
ACK 40c7943e3abacff0a9ead1cd4cf20d4b763496d3 BCPRRC edits

jnewbery commented:
ACK. Another smashing newsletter! Thanks team :hugs:

bitschmidty commented:
5 different authors this week 🙌 

Thanks to @harding @jonatack @jnewbery @dongcarl ! @vasild thanks for the review!
